ktor-features/ktor-velocity/jvm/src/io/ktor/velocity/VelocityTools.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.velocity
+
+import io.ktor.application.*
+import io.ktor.response.*
+import io.ktor.util.*
+import org.apache.velocity.app.*
+import org.apache.velocity.tools.*
+import org.apache.velocity.tools.config.*
+
+private val ENGINE_CONFIG_KEY = "ENGINE_CONFIG"
+
+public fun EasyFactoryConfiguration.engine(configure : VelocityEngine.() -> Unit) {
+    data(ENGINE_CONFIG_KEY, configure)
+}
+
+/**
+ * VelocityTools ktor feature. Populates model with standard Velocity tools.
+ */
+public class VelocityTools(private val toolManager: ToolManager) {
+
+    /**
+     * A companion object for installing feature
+     */
+    public companion object Feature :
+        ApplicationFeature<ApplicationCallPipeline, EasyFactoryConfiguration, VelocityTools> {
+
+        override val key: AttributeKey<VelocityTools> = AttributeKey<VelocityTools>("velocityTools")
+
+        override fun install(
+            pipeline: ApplicationCallPipeline,
+            config: EasyFactoryConfiguration.() -> Unit
+        ): VelocityTools {
+            val factoryConfig = EasyFactoryConfiguration().apply(config)
+            val engineConfig = factoryConfig.getData(ENGINE_CONFIG_KEY)
+            factoryConfig.removeData(engineConfig)
+            val engine = VelocityEngine().apply(engineConfig.value as VelocityEngine.() -> Unit)
```

Comment:
Should we check if the engine was set in config? Maybe fallback to the default engine otherwise? And can you add test for missing engine config, please?

Reply:
The engine config should definitely be optional. Done, with its test case. The doc shall contain a link towards the default engine config (btw, I'll make a PR for the docs once we're good here).